pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -273,15 +277,15 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
 	}
 
 	log.Infof("doBaseOsActivate: %s activating", uuidStr)
-	zboot.SetOtherPartitionStateUpdating(log)
-	publishZbootPartitionStatus(ctx, status.PartitionLabel)
-	baseOsSetPartitionInfoInStatus(ctx, status, status.PartitionLabel)
-	publishBaseOsStatus(ctx, status)
 
 	// install the image at proper partition; dd etc
-	if installDownloadedObjects(uuidStr, status.PartitionLabel,
-		&status.ContentTreeStatusList) {
-
+	changed, proceed, err = installDownloadedObjects(uuidStr, status.PartitionLabel,
+		&status.ContentTreeStatusList)
+	if err != nil {
+		status.SetErrorNow(err.Error())
+		changed = true
+		return changed
+	} else if proceed {
```

Comment:
Nit: Since you have a return in the if,having an else if doesn't improve readability.

Reply:
Actually it would be clearer if it said:
if err != nil {
   ...
   return
}
if !proceed {
   ...
   return
}

Reply:
Resolved